table/block_based_table_reader.cc

```diff
@@ -2193,6 +2197,7 @@ void BlockBasedTableIterator<TBlockIter>::FindKeyForward() {
 
 template <class TBlockIter>
 void BlockBasedTableIterator<TBlockIter>::FindKeyBackward() {
+  assert(!is_out_of_bound_);
```

Comment:
I am curious about why we don't need to check for the lower bound at the end of this function. 
Would we be missing any out-of-bounds cases by not having that check? Or is there some other place where the lower bound check is done?

Reply:
At this level, it is just the optimization. The final out-of-bound check will be don in DBIter, so there isn't a correctness issue whether we check it or not here. I just didn't want to complicate the code, worrying bugs like this.